### PR TITLE
feat: resource details added to findigns and resource view

### PR DESCRIPTION
**Pull request by alejandrobailo**
### Context
  Adds missing resource details (partition, details, metadata) to the Findings and Resources detail
  views to provide more comprehensive resource information.

  ### Description
  - Added **Partition**, **Details**, and **Metadata** fields to resource detail displays
  - Updated TypeScript types in `types/components.ts` and `types/resources.ts`
  - Metadata rendered as key-value pairs with proper object serialization

  **Note:** Remove debug console.log statements before merge.

  ### Steps to review
  1. **Type Safety** - Review type changes for new nullable fields (`partition`, `details`, `metadata`)
  2. **Findings View** - Navigate to any finding detail and verify new Partition/Details fields display
   correctly
  3. **Resources View** - Check resource detail page shows Partition, Details, and Metadata section
  (when available)
  4. **Edge Cases** - Test with resources lacking these fields (should show "--")
  5. **Code Quality**
     - Remove console.log statements from `finding-detail.tsx:58-59` and `resource-detail.tsx:86-90`
     - Run `npm run typecheck` and `npm run lint:check`
     
     
Screenshot 2025-12-11 at 15 12 48
Screenshot 2025-12-11 at 15 13 20
Screenshot 2025-12-11 at 17 35 50
Screenshot 2025-12-11 at 17 36 00

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.
- [ ] Review if is needed to change the [Readme.md](https://github.com/prowler-cloud/prowler/blob/master/README.md)
- [ ] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/prowler/CHANGELOG.md), if applicable.

#### UI
- [X] All issue/task requirements work as expected on the UI
- [X] Screenshots/Video of the functionality flow (if applicable) - Mobile (X < 640px)
- [X] Screenshots/Video of the functionality flow (if applicable) - Table (640px > X < 1024px)
- [X] Screenshots/Video of the functionality flow (if applicable) - Desktop (X > 1024px)
- [X] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/ui/CHANGELOG.md), if applicable.

#### API
- [ ] Verify if API specs need to be regenerated.
- [ ] Check if version updates are required (e.g., specs, Poetry, etc.).
- [ ] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/api/CHANGELOG.md), if applicable.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ui/CHANGELOG.md [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
+## [1.15.0] (Prowler Unreleased)
 
 ### 🚀 Added
 
 - Risk Plot component with interactive legend and severity navigation to Overview page [(#9469)](https://github.com/prowler-cloud/prowler/pull/9469)
 - Navigation progress bar for page transitions using Next.js `onRouterTransitionStart` [(#9465)](https://github.com/prowler-cloud/prowler/pull/9465)
-- Findings Severity Over Time chart component to Overview page [(#9405)](https://github.com/prowler-cloud/prowler/pull/9405)
+- Finding Severity Over Time chart component to Overview page [(#9405)](https://github.com/prowler-cloud/prowler/pull/9405)
 - Attack Surface component to Overview page [(#9412)](https://github.com/prowler-cloud/prowler/pull/9412)
+- Resource details (partition, details and metadata) [(#9515)](https://github.com/prowler-cloud/prowler/pull/9515)
```
Comment (jfagoagas) [inline ui/CHANGELOG.md]: v1.16.0 🥇

**Review by jfagoagas — APPROVED** [review]
